libsolidity/formal/SMTEncoder.cpp

```diff
@@ -932,7 +932,26 @@ void SMTEncoder::visitObjectCreation(FunctionCall const& _funCall)
 void SMTEncoder::endVisit(Identifier const& _identifier)
 {
 	if (auto decl = identifierToVariable(_identifier))
-		defineExpr(_identifier, currentValue(*decl));
+	{
+		if (decl->isConstant())
+		{
+			if (RationalNumberType const* rationalType = isConstant(_identifier))
+			{
+				if (rationalType->isNegative())
+					defineExpr(_identifier, smtutil::Expression(u2s(rationalType->literalValue(nullptr))));
+				else
+					defineExpr(_identifier, smtutil::Expression(rationalType->literalValue(nullptr)));
+			}
+			else
```

Comment:
Which case is this? Can you add a test for it?

Reply:
There is already a test for it, that's where it broke before this.
`string constant S = "abc";`

Reply:
`test/libsolidity/smtCheckerTests/complex/slither/const_state_variables.sol`